src/app/public/src/modules/table-of-contents/table-of-contents.component.ts

```diff
@@ -9,4 +10,53 @@ import { StacheNavLink } from '../nav';
 export class StacheTableOfContentsComponent {
   @Input()
   public routes: StacheNavLink[];
+  private viewTop: number;
+  private documentBottom: number;
+
+  constructor(
+    private windowRef: StacheWindowRef,
+    private omnibarService: StacheOmnibarAdapterService
+  ) {
+    this.windowRef.scrollEvent.subscribe(() => {
+      this.updateRoutesOnScroll(this.routes);
+    });
+  }
+
+  public updateRoutesOnScroll(routes: any[]) {
+    if (routes && routes.length) {
+      this.updateView(routes);
+    }
+  }
+
+  // Updates ng-class level trigger to show/hide blue border marking location on page
+  public updateView(routes: StacheNavLink[]) {
+    this.trackViewTop();
+    this.isCurrent(routes);
+  }
+
+  private trackViewTop() {
+    this.viewTop = (this.windowRef.nativeWindow.pageYOffset + this.omnibarService.getHeight());
+    this.documentBottom = Math.round(this.windowRef.nativeWindow.document.documentElement.getBoundingClientRect().bottom);
+  }
+
+  private isCurrent(routes: StacheNavLink[]): void {
+    if (this.scrollEnd()) {
```

Comment:
scrollEnd can be read as ending the scroll, not necessarily hitting the bottom of the page.  Should this be named something more telling?